src/Ocelot/DownstreamUrlCreator/Middleware/DownstreamUrlCreatorMiddleware.cs

```diff
@@ -1,156 +1,179 @@
+        public DownstreamUrlCreatorMiddleware(RequestDelegate next,
+            IOcelotLoggerFactory loggerFactory,
+            IDownstreamPathPlaceholderReplacer replacer
+            )
+                : base(loggerFactory.CreateLogger<DownstreamUrlCreatorMiddleware>())
+        {
+            _next = next;
+            _replacer = replacer;
+        }
 
-using Ocelot.Logging;
+        public async Task Invoke(HttpContext httpContext)
+        {
+            var downstreamRoute = httpContext.Items.DownstreamRoute();
 
-using Microsoft.AspNetCore.Http;
+            var templatePlaceholderNameAndValues = httpContext.Items.TemplatePlaceholderNameAndValues();
 
-using Ocelot.Middleware;
-using Ocelot.Request.Middleware;
+            var response = _replacer
+                .Replace(downstreamRoute.DownstreamPathTemplate.Value, templatePlaceholderNameAndValues);
 
-using Ocelot.Responses;
+            var downstreamRequest = httpContext.Items.DownstreamRequest();
 
-using Ocelot.DownstreamUrlCreator.UrlTemplateReplacer;
+            if (response.IsError)
+            {
+                Logger.LogDebug("IDownstreamPathPlaceholderReplacer returned an error, setting pipeline error");
+
+                httpContext.Items.UpsertErrors(response.Errors);
+                return;
+            }
+
+            if (!string.IsNullOrEmpty(downstreamRoute.DownstreamScheme))
+            {
+                //todo make sure this works, hopefully there is a test ;E
+                httpContext.Items.DownstreamRequest().Scheme = downstreamRoute.DownstreamScheme;
+            }
+
+            var internalConfiguration = httpContext.Items.IInternalConfiguration();
+
+            if (ServiceFabricRequest(internalConfiguration, downstreamRoute))
+            {
+                var pathAndQuery = CreateServiceFabricUri(downstreamRequest, downstreamRoute, templatePlaceholderNameAndValues, response);
+
+                //todo check this works again hope there is a test..
+                downstreamRequest.AbsolutePath = pathAndQuery.Path;
+                downstreamRequest.Query = pathAndQuery.Query;
+            }
+            else
+            {
+                var dsPath = response.Data;
+
+                if (ContainsQueryString(dsPath))
+                {
+                    downstreamRequest.AbsolutePath = GetPath(dsPath);
+
+                    if (string.IsNullOrEmpty(downstreamRequest.Query))
+                    {
+                        downstreamRequest.Query = GetQueryString(dsPath);
+                    }
+                    else
+                    {
+                        //downstreamRequest.Query += GetQueryString(dsPath).Replace('?', '&');
+                        var newQueryString = GetQueryString(dsPath).Replace('?', '&');
+                        newQueryString = MergeQueryStringsWithoutDuplicateValues(downstreamRequest.Query, newQueryString);
+                        downstreamRequest.Query = "?" + newQueryString;                    }
```

Comment:
```suggestion
                        downstreamRequest.Query = '?' + newQueryString;                    }
```

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/1182/commits/275d034250762557f50e8461d27b3446893b54a1